picfixPy/sharpen.py

```diff
@@ -1,20 +1,112 @@
     intensity -- integer in range [0,10] to specificy intensity of sharpening applied to the image,
                  defaults to 5.
-    output_img -- string path for sharpened .png image
+    display -- bool. If True, output will be displayed. Defaults to False.
+    output_img -- string path for output of corresponding .jpg, .jpeg, .png image with adjusted sharpening 
     
     Returns
     ============
-    an sharpened image file in .png format in specified output path
+    an image file in .jpg, .jpeg, .png format with adjusted sharpening intensity in specified output path
     
     Example
     ============
-    sharpen("../img/image.png", 2, "../img/sharpened_image.png")
-    '''
\ No newline at end of file
+    sharpen("../img/image.jpg", 5, display=False, "../img/sharpened_image.jpg")
+    '''
+
+    # load input image
+    # lift pixel value restrictions between 0 and 255 temporarily
+    try: 
+        input_img = skimage.io.imread(input_img).astype(int)
+    
+    except AttributeError:
+        print("Please use a string to specify the file path for an input image.")
+        raise
+    
+    except OSError:
+        print("Please provide an image file.")
+        raise
+    
+    except FileNotFoundError:
+        print("Cannot find image file.")
+            
+    except Exception as error:
+        print("Cannot load image, something went wrong :(")
+        print(error)
+        raise
+        
+    # validate intensity level
+    if (intensity < 0 or intensity > 10):
+        raise ValueError("Intensity level must be between 0 and 10.")
+        
+        
+    # Guassian blurring of image
+    
+    # approximation of Gaussian blur 3x3 kernel
+    # kernel attribution: https://en.wikipedia.org/wiki/Kernel_(image_processing)
+    kernel = np.array([[1.0,2.0,1.0], [2.0,4.0,2.0], [1.0,2.0,1.0]])
+    kernel = kernel / np.sum(kernel)
+    
+    # attribution: https://bit.ly/2EbS4XD
+    image_array = []
+    for y in range(3):
+        temp_array = np.copy(input_img)
+        temp_array = np.roll(temp_array, y - 1, axis=0)
+        for x in range(3):
+            temp_array_X = np.copy(temp_array)
+            temp_array_X = np.roll(temp_array_X, x - 1, axis=1)*kernel[y,x]
+            image_array.append(temp_array_X)
+
+    image_array = np.array(image_array)
+    image_array_sum = np.sum(image_array, axis=0)
+    
+    
+    # sharpening of image by unsharp mask
+    
+    # https://en.wikipedia.org/wiki/Unsharp_masking
+    # https://homepages.inf.ed.ac.uk/rbf/HIPR2/unsharp.htm
+    
+    # blurred image 
+    blurred = image_array_sum
+    
+    # concept: sharpened = original + (original − blurred) × amount.
+    # attribution: https://bit.ly/2V3FAr0
+    sharpened = float(intensity + 1) * input_img - float(intensity) * blurred
+    sharpened = np.maximum(sharpened, np.zeros(sharpened.shape))
+    sharpened = np.minimum(sharpened, 255 * np.ones(sharpened.shape))
+    sharpened = sharpened.round().astype(np.uint8)
```

Comment:
Nice vectorized way of restricting the value of the pixel back to between 0 and 255, and rounding it back to an integer. Nicely done.